components/layout/context.rs

```diff
@@ -120,31 +120,21 @@ impl<'a> LayoutContext<'a> {
         };
 
         // See if the image is already available
-        let result =
-            self.image_cache
-                .find_image_or_metadata(url.clone(), use_placeholder, can_request);
-        match result {
-            Ok(image_or_metadata) => Some(image_or_metadata),
-            // Image failed to load, so just return nothing
-            Err(ImageState::LoadError) => None,
-            // Not yet requested - request image or metadata from the cache
-            Err(ImageState::NotRequested(id)) => {
-                let image = PendingImage {
-                    state: PendingImageState::Unrequested(url),
-                    node: node.to_untrusted_node_address(),
-                    id: id,
-                };
-                self.pending_images
-                    .as_ref()
-                    .unwrap()
-                    .lock()
-                    .unwrap()
-                    .push(image);
-                None
-            },
+        match self.image_cache.get_image(&url, use_placeholder) {
+            Some(x) => return Some(ImageOrMetadataAvailable::ImageAvailable(x, url)),
```

Comment:
How about using `if let` here 👀? then we don't need to do nothing with `()` for wildcard arm.

```rust
if let Some(x) = self.image_cache.get_image(&url, use_placeholder {
  return Some(ImageOrMetadataAvailable::ImageAvailable(x, url));
}
```

Reply:
totally, that makes way more sense! still not used to being able to do that, so cool

Reply:
btw, I see that one can add a semicolon when returning, but it looks like it compiled also without the semicolon, which is what I assumed since you're returning the `Option` - is the return with a semicolon idiomatic rust?